R/nested_model_fit.R

```diff
@@ -0,0 +1,134 @@
+#' Axing a nested_model_fit.
+#'
+#' nested_model_fit objects are created from the \pkg{nestedmodels}
+#' package, which allows parsnip models to be fitted on nested data. Axing a
+#' nested_model_fit object involves axing all the inner model_fit objects.
+#'
+#' @inheritParams butcher
+#'
+#' @seealso [axe-model_fit]
+#'
+#' @return Axed nested_model_fit object.
+#'
+#' @examplesIf rlang::is_installed(c("parsnip", "nestedmodels"))
+#'
+#' library(nestedmodels)
+#' library(parsnip)
+#'
+#' model <- linear_reg() %>%
+#'   set_engine("lm") %>%
+#'   nested()
+#'
+#' nested_data <- tidyr::nest(example_nested_data, data = -id)
+#'
+#' fit <- fit(model, z ~ x + y + a + b, nested_data)
+#'
+#' # Reduce the model size
+#' butcher(fit)
+#'
+#' @name axe-nested_model_fit
+NULL
+
+#' Remove the call.
+#'
+#' @rdname axe-nested_model_fit
+#' @export
+axe_call.nested_model_fit <- function(x, verbose = FALSE, ...) {
+  old <- x
+  x$fit$.model_fit <- purrr::map(
+    x$fit$.model_fit,
+    axe_call,
+    verbose = FALSE,
+    ...
+  )
+
+  all_disabled <- purrr::map(x$fit$.model_fit, attr, "disabled")
+  disabled <- unique(purrr::list_c(all_disabled, ptype = character()))
+  if (length(disabled) == 0) {
+    disabled <- NULL
+  }
+  add_butcher_attributes(x, old, disabled = disabled, verbose = verbose)
+}
+
+#' Remove controls used for training.
+#'
+#' @rdname axe-nested_model_fit
+#' @export
+axe_ctrl.nested_model_fit <- function(x, verbose = FALSE, ...) {
+  old <- x
+  x$fit$.model_fit <- purrr::map(
+    x$fit$.model_fit,
+    axe_ctrl,
+    verbose = FALSE,
```

Comment:
```suggestion
    verbose = verbose,
```
Let me know if I am misunderstanding this